### ISSUE TITLE
[FEATURE] Categorical Values in Random-Forest

### ISSUE THREAD

**Issue by Deathn0t** [deephyper/deephyper#174]
**Is your feature request related to a problem? Please describe.**

The current implementation of Random-Forest is based on `scikit-learn` with a different `splitter`. See [deephyper.skopt.learning.forest](https://github.com/deephyper/deephyper/blob/master/deephyper/skopt/learning/forest.py#L68). 

When we have Categorical variables, initially we would one-hot encode them but it reduces a lot the scalability of the method because increase quickly the feature-dimensionality of the learning algorithm. Therefore we encode them as integer labels.

But, this make the splitting decision for categorical variables sub-optimal.

**Describe the solution you'd like**

Categorical values should be handle with best-sub-sets selection instead. Some example implementations are available at:
* [pyrfr Python package](https://github.com/automl/random_forest_run)
* [sklearn HistGradientBoostingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html)

When implementing this feature, we should be careful about the uncertainty quantification (`splitter="random"`) in many implementations HistGradientBoostingRegressor the `splitter="best"` is used for example.
